src/high-level/scripts/papi_hl_output_writer.py

```diff
@@ -75,10 +75,45 @@ def merge_json_files(source):
   
   json_object['ranks'] = json_rank
 
-  # print json.dumps(json_object,indent=2, sort_keys=False,
-  #                  separators=(',', ': '), ensure_ascii=False)
   return json_object
 
+def parse_source_file(source_file):
+    """Take JSON file (source_file) and convert it to a python dictionary.
```

Comment:
I added a Python doc string to the function I created titled `parse_source_file`, I think adding them to the other individual functions could be beneficial for two reasons:

1. Adding the doc strings can help speed up development. A doc string would help with understanding what each function is doing rather than having to walk through a function with no prior background.
2. If a user wanted to look at the contents of a function without going to the actual file they could run the following: 
```
# code 
from papi_hl_output_writer.py import parse_source_file
help(parse_source_file)
# output
Take a JSON file (source_file) and convert it to a python dictionary.
Additionally add extra metadata such as mpi rank.
    
Arguments:
    source_file -- JSON file generated from PAPI high level API functions.
Returns:
    Python dictionary.
```

This is nothing major, but I wanted to mention it.

Reply:
It does look possible to add `doxygen` to a Python file, I will work on creating an example.